lib/core/src/plugin.rs

```diff
@@ -0,0 +1,9 @@
+use maybe_sync::{MaybeSend, MaybeSync};
+
+use crate::sdk::LiquidSdk;
+use sdk_common::utils::Arc;
+
+pub trait Plugin: MaybeSend + MaybeSync {
+    fn on_start(&self, sdk: Arc<LiquidSdk>);
```

Comment:
From the nwc PR I think I noticed the plugin uses storage. Shouldn't we add here a key value store trait and pass it to on_start?